### PR TITLE
Feature/jet stream

### PR DESCRIPTION
**Pull request by boecklim**
- Optionally allow to use NATS JetStream client instead of NATS Core client
- Refactor & rename async package and configuration
- Integration test for NATS JetStream client
- Upgrade go version to 1.22.5 => integration test coverage was ignored before

### CODE REVIEW

**Comment by shotasilagadzetaal** [comment]
I think we should be merging into staging first from now on? 🤔

File: internal/message_queue/nats/nats_connection/nats_connection.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			}
+		}),
+		nats.ReconnectHandler(func(_ *nats.Conn) {
+			logger.Info("client reconnected")
+		}),
+		nats.ClosedHandler(func(_ *nats.Conn) {
+			logger.Info("client closed")
+		}),
+		nats.RetryOnFailedConnect(true),
+		nats.PingInterval(2 * time.Minute),
```
Comment (shotasilagadzetaal) [inline internal/message_queue/nats/nats_connection/nats_connection.go:43]: Isn't two minutes too long for ping interval?

File: internal/message_queue/nats/client/nats_jetstream/nats_jetstream_client.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+	js, err := jetstream.New(nc)
+	if err != nil {
+		return nil, err
+	}
+
+	p.js = js
+
+	for _, topic := range topics {
+		_, err = p.getStream(topic, fmt.Sprintf("%s-stream", topic))
```
Comment (shotasilagadzetaal) [inline internal/message_queue/nats/client/nats_jetstream/nats_jetstream_client.go:78]: What's the point of this call? Just checking if we could encounter error?

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/bitcoin-sv/arc
 
-go 1.21.3
+go 1.22.5
```
[the remark this answers is not in the document]
Comment (boecklim) [reply go.mod:3]: Yes, that was a mistake. Will change the description.

File: internal/message_queue/nats/client/nats_jetstream/nats_jetstream_client.go (hunk 2 above)
Comment (boecklim) [reply internal/message_queue/nats/client/nats_jetstream/nats_jetstream_client.go:78]: The `getStream()` function returns the stream, but also ensures it exists i.e. it returns the existing one or creates it if it not exists. For a pure publisher the stream is not needed. It is only needed for a subscriber. Therefore there is no stream returned.

**Review by kuba-4chain — APPROVED** [review]

**Pull request merged**